### DISCUSSION TITLE
nautical NOAA charts

### DISCUSSION THREAD

**Discussion by dkerlee** [osmandapp/OsmAnd#12412] [Maps Related]
curious if anything has been, or can be done, about integrating free NOAA charts with OSMAND. I looked at the nautical chart section and it's pretty weak-sauce. Both raster and vector charts are available for for free from NOAAs website:
https://www.charts.noaa.gov/ENCs/ENCs.shtml

**Comment by vshcherb** [comment]
Interesting question, cc @xmd5a2.

**Comment by quantenschaum** [comment]
The [online maps plugin](https://osmand.net/docs/user/plugins/online-map) allows you to add custom raster maps to OsmAnd. These tiles can be predownloaded for offline use. So, if you can turn the ENC into tile images and put them on a web server, you can use them in OsmAnd. [QGIS](https://www.qgis.org/) can be used to generate map tiles.

A similar approach has been taken for chart data available for the Netherlands: https://github.com/quantenschaum/mapping This can be adapted for the use with the NOAA charts.

**Comment by quantenschaum** [comment]
There are some servers out there that already provide the data from the NOAA charts as tiles. They can easily be added to OsmAnd using the [magic URL](https://osmand.net/docs/user/map/raster-maps#magic-url-to-install-map-source) if they do provide slippy tiles.

- https://www.siitech.com/ runs a tiles server [https://charts.siitech.com:494/tileNoaa4/{0}/{1}/{2}.png?c=1&m=2&w=1](http://osmand.net/add-tile-source?name=SIITECH&min_zoom=2&max_zoom=18&url_template=https%3A%2F%2Fcharts.siitech.com%3A494%2FtileNoaa4%2F%7B0%7D%2F%7B1%7D%2F%7B2%7D.png%3Fc%3D1%26m%3D2%26w%3D1)
- The US Navy makes the charts available as [WMS](https://en.wikipedia.org/wiki/Web_Map_Service) at https://geoint.nrlssc.navy.mil/tsclient/. 
- NOAA also runs a WMS providing [DNC](https://gis.charttools.noaa.gov/arcgis/rest/services/MarineChart_Services/NOAACharts/MapServer) and [ENC](https://gis.charttools.noaa.gov/arcgis/rest/services/MCS/ENCOnline/MapServer)
- Canada provides a similar service also serving US charts https://e-navigation.canada.ca/interactive-map

Using https://mapproxy.org/ it is possible to turn this into a [TMS](https://en.wikipedia.org/wiki/Tile_Map_Service)/[Slippy Tiles](https://en.wikipedia.org/wiki/Tiled_web_map) as required by OsmAnd online maps plugin. Some web viewers use slippy tiles, you can extract the URLs using your browser add them to OsmAnd.
